fire/inspectutils.py

```diff
@@ -107,6 +107,9 @@ def GetFullArgSpec(fn):
   if skip_arg and args:
     args.pop(0)  # Remove 'self' or 'cls' from the list of arguments.
 
+  if args and args[0] == 'self':
```

Comment:
This should already be handled by 108.
If it wasn't, then skip_arg wasn't being set to True somewhere where it should have been set to True.

What example caused you to add this check?

Reply:
The failing `testFireKeywordArgs` and `testFireObjectWithDict` cases did.

Maybe we should add the check 
`skip_arg = inspect.isbuiltin(fn) and fn.__self__ is not None` in `_GetArgSpecInfo` or even simpler, just check `fn.__self__ is not None` and ignore if it is a method or not?

Here was my earlier analysis:
> The `testFireKeywordArgs` and `testFireObjectWithDict` cases are more subtle than that - the specific return value of `inspect.getargspec()` is `FullArgSpec(args=['self'], varargs=None, varkw=None, defaults=None, kwonlyargs=[], kwonlydefaults=None, annotations={})`, in particular, that `self` in `args` is causing the test cases to expect an extra parameter and leading to subsequent failures.

Reply:
I don't remember offhand why the ismethod check is needed, but I suspect there was a reason. (If we figure out that reason, let's add a comment saying what it is.)

Your idea of augmenting _GetArgSpecInfo so that skip_arg becomes True for the builtins that take self sgtm.

Reply:
Just pushed a new version using the idea.

Reply:
Looks good to me 👍